sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java

```diff
@@ -78,7 +82,7 @@ private void tryForwardResult(Throwable throwable) {
         }
         if (failure == null) {
             assert iterator != null : "iterator must be present";
-            consumer.accept(iterator, null);
+            executor.execute(() -> consumer.accept(iterator, null));
```

Comment:
should we handle the rejectedexecutionexception exceptions here? (maybe even use our `CompletableFutures.supplyAsync`)

Reply:
See https://github.com/crate/crate/pull/7853/files#diff-dd02b03e4e7ce6a1e8bb0edec0bde6ceR66 - the Executor has a fallback. 
Maybe that's a bit too subtle. I can also change it to have explicit handling here?

Reply:
Merged it for now, I can follow up with another PR to make it explicit if you want, but for now I'd like to get the flakyness away to unblock the other PRs

Reply:
Ah right, that's fine